server/methods/core/orders.js

```diff
@@ -974,51 +974,57 @@ export const methods = {
       // Grab the amount from the shipment, otherwise use the original amount
       const processor = paymentMethod.processor.toLowerCase();
 
-      Meteor.call(`${processor}/payment/capture`, paymentMethod, (error, result) => {
-        if (result && result.saved === true) {
-          const metadata = Object.assign(bilingRecord.paymentMethod.metadata || {}, result.metadata || {});
-
-          Orders.update({
-            "_id": orderId,
-            "billing.paymentMethod.transactionId": transactionId
-          }, {
-            $set: {
-              "billing.$.paymentMethod.mode": "capture",
-              "billing.$.paymentMethod.status": "completed",
-              "billing.$.paymentMethod.metadata": metadata
-            },
-            $push: {
-              "billing.$.paymentMethod.transactions": result
-            }
-          });
+      let result;
+      let error;
+      try {
+        result = Meteor.call(`${processor}/payment/capture`, paymentMethod);
+      } catch (e) {
+        error = e;
+      }
```

Comment:
The only change here is that the Meteor.call of `${processor}/payment/capture` is now synchronous (has no callback). Before this change, I had tests failing because that call would still be executing after the test had already finished and removed the sandbox data. Doing it synchronous seems like it would be better in real world, too, but on the off chance that this was intentionally async, there might be a different solution by rewriting how the tests work.